cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/consumption/AwsS3ConsumptionCalculator.java

```diff
@@ -0,0 +1,105 @@
+import com.sequenceiq.cloudbreak.cloud.ConsumptionCalculator;
+import com.sequenceiq.cloudbreak.cloud.aws.common.connector.resource.AwsCloudWatchCommonService;
+import com.sequenceiq.cloudbreak.cloud.exception.CloudConnectorException;
+import com.sequenceiq.cloudbreak.cloud.model.CloudConsumption;
+import com.sequenceiq.cloudbreak.cloud.model.objectstorage.ObjectStorageSizeRequest;
+import com.sequenceiq.cloudbreak.cloud.model.objectstorage.ObjectStorageSizeResponse;
+import com.sequenceiq.cloudbreak.common.mappable.CloudPlatform;
+import com.sequenceiq.cloudbreak.common.mappable.StorageType;
+import com.sequenceiq.common.model.FileSystemType;
+
+@Service
+public class AwsS3ConsumptionCalculator implements ConsumptionCalculator {
+
+    private static final StorageType S3 = StorageType.S3;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsS3ConsumptionCalculator.class);
+
+    private static final int DATE_RANGE_WIDTH_IN_DAYS = 2;
+
+    private static final long NO_BYTE_IN_MB = 1000L * 1000L;
+
+    @Inject
+    private AwsCloudWatchCommonService cloudWatchCommonService;
+
+    @Override
+    public void validate(CloudConsumption cloudConsumption) throws ValidationException {
+        String storageLocation = cloudConsumption.getStorageLocation();
+        if (storageLocation == null || !storageLocation.startsWith(FileSystemType.S3.getProtocol())) {
+            throw new ValidationException(String.format("Storage location must start with '%s' if required file system type is '%s'!",
+                    FileSystemType.S3.getProtocol(), FileSystemType.S3.name()));
+        }
+    }
+
+    @Override
+    public ObjectStorageSizeResponse calculate(ObjectStorageSizeRequest request) {
+        GetMetricStatisticsResult result = cloudWatchCommonService.getBucketSize(request.getCredential(), request.getRegion().value(),
+                request.getStartTime(), request.getEndTime(), request.getObjectStoragePath());
+        Optional<Datapoint> latestDatapoint = result.getDatapoints().stream().max(Comparator.comparing(Datapoint::getTimestamp));
+        if (latestDatapoint.isPresent()) {
+            Datapoint datapoint = latestDatapoint.get();
+            LOGGER.debug("Gathered datapoint from CloudWatch: {}", datapoint);
+            return ObjectStorageSizeResponse.builder().withStorageInBytes(datapoint.getMaximum()).build();
+        } else {
+            String message = String.format("No datapoints were returned by CloudWatch for bucket %s and timeframe from %s to %s",
+                    request.getObjectStoragePath(), request.getStartTime().toString(), request.getEndTime().toString());
+            LOGGER.error(message);
+            throw new CloudConnectorException(message);
+        }
+    }
+
+    @Override
+    public String getObjectId(String objectId) {
+        objectId = objectId.replace(FileSystemType.S3.getProtocol() + "://", "");
+        return objectId.split("/")[0];
+    }
```

Comment:
@doktoric  suggested to change the name because for example for EBS it will be Volume ID and for EFS it can be something else so it is not only **storageLocation**